enhancements/project-request-limit.md

```diff
@@ -0,0 +1,345 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+The `requestlimit.project.openshift.io/ProjectRequestLimit` admission plugin is used to impose a limit on the number of
+self-provisioned project(s) requested by a given user. In 3.x, the limit can be specified via the master configuration file.
+It doesn't fit well with our targeted configuration which aims to avoid adding lots of intricately documented knobs. 
+This is why this admission plug-in is disabled in OpenShift 4.x. 
+
+Instead of wiring the admission plugin via the `kube-apiserver` operator, we can create a validating admission webhook 
+based on the [generic-admission-server](https://github.com/openshift/generic-admission-server) and distribute it via
+OperatorHub/OLM.
+
+## Motivation
+* Some customers use the `ProjectRequestLimit` plugin in 3.x and thus are blocked on their path to OpenShift 4.x. Enabling
+ it on 4.x will unblock these customers.
+* The `kube-apiserver` is designed to be extended using validating/mutating admission webhook(s), we have the
+technology to easily build one, and we have the ability to create a simple operator to manage it.
+* The cluster admin can optionally enable the validating webhook on a cluster and configure limits via a custom
+resource. This will sever the "legacy" link between the plugin and the `kube-apiserver`.
+* We can provide the customer with seamless install and automatic upgrades by shipping the operator via OperatorHub 
+  independent of OpenShift release cycle.   
+
+
+### Goals
+1. Enable the `requestlimit.project.openshift.io/ProjectRequestLimit` admission plugin.
+2. Use existing extension points, libraries, and installation mechanisms in the manner we would recommend to external teams.
+3. Have a fairly straightforward way to install and enable this admission plugin.
+
+### Non-Goals
+1. Revisit how `requestlimit.project.openshift.io/ProjectRequestLimit` works. We want to lift it as is.
+2. Couple a slow moving admission plugin to a fast moving `kube-apiserver`.
+3. Since `Project` type is owned by `openshift-apiserver`, rebootstrapping is not applicable.
+
+### Open Questions
+1. How do we productize the operator? Since this operator is not part of OpenShift core, it can essentially be built and
+ pushed to OperatorHub independently of OpenShift release cycle.
+2. How do we describe version skew limitations to OLM so our operator gets uninstalled *before* an illegal downgrade or
+ upgrade? This is a concrete case of the API we want to use isn't available before 1.16 and after 1.18, the previous API could be gone.
+
+## Proposal
+1. Create a `Validating` admission webhook server that provides `requestlimit.project.openshift.io/ProjectRequestLimit`.
+2. The admission webhook will be fronted by a `Service`.
+3. `kube-apiserver` will connect to the admission webhook server via the `kubernetes.default.svc service` service.
```

Comment:
openshift-apiserver connects to kube-apiserver through that service. The kube-apiserver connects to the webhook via the service in (2)

Reply:
I clarified a bit more, please take a look.